nsgt/fft.py

```diff
@@ -14,102 +14,109 @@
-                return (plan,None,None)
-    
-        class rfftp(fftpool):
-            def __init__(self, measure=False, dtype=float):
-                fftpool.__init__(self, measure, dtype=dtype)
-            def init(self, n, measure, outn):
-                inp = self.fftw.create_aligned_array(n, dtype=self.tpfloat)
-                outp = self.fftw.create_aligned_array(n//2+1, dtype=self.tpcplx)
-                plan = self.fftw.Plan(inp, outp, direction='forward', realtypes='halfcomplex r2c',flags=('measure' if measure else 'estimate',))
-                return (plan,None,None)
-    
-        class ifftp(fftpool):
-            def __init__(self, measure=False, dtype=float):
-                fftpool.__init__(self, measure, dtype=dtype)
-            def init(self, n, measure, outn):
-                inp = self.fftw.create_aligned_array(n, dtype=self.tpcplx)
-                outp = self.fftw.create_aligned_array(n, dtype=self.tpcplx)
-                plan = self.fftw.Plan(inp, outp, direction='backward', flags=('measure' if measure else 'estimate',))
-                return (plan,None,lambda x: x/len(x))
-    
-        class irfftp(fftpool):
-            def __init__(self, measure=False, dtype=float):
-                fftpool.__init__(self, measure, dtype=dtype)
-            def init(self, n, measure, outn):
-                inp = self.fftw.create_aligned_array(n, dtype=self.tpcplx)
-                outp = self.fftw.create_aligned_array(outn if outn is not None else (n-1)*2, dtype=self.tpfloat)
-                plan = self.fftw.Plan(inp, outp, direction='backward', realtypes='halfcomplex c2r', flags=('measure' if measure else 'estimate',))
-                return (plan,lambda x: x[:n],lambda x: x/len(x))
-            
-        realized = True
-        
+    class ifftp(fftpool):
+        def __init__(self, measure=False, dtype=float):
+            fftpool.__init__(self, measure, dtype=dtype)
+        def init(self, n, measure, outn):
+            inp = self.fftw.create_aligned_array(n, dtype=self.tpcplx)
+            outp = self.fftw.create_aligned_array(n, dtype=self.tpcplx)
+            plan = self.fftw.Plan(inp, outp, direction='backward', flags=('measure' if measure else 'estimate',))
+            return (plan,None,lambda x: x/len(x))
 
-if not realized:
+    class irfftp(fftpool):
+        def __init__(self, measure=False, dtype=float):
+            fftpool.__init__(self, measure, dtype=dtype)
+        def init(self, n, measure, outn):
+            inp = self.fftw.create_aligned_array(n, dtype=self.tpcplx)
+            outp = self.fftw.create_aligned_array(outn if outn is not None else (n-1)*2, dtype=self.tpfloat)
+            plan = self.fftw.Plan(inp, outp, direction='backward', realtypes='halfcomplex c2r', flags=('measure' if measure else 'estimate',))
+            return (plan,lambda x: x[:n],lambda x: x/len(x))
+elif pyfftw is not None:
+    ENGINE = "PYFFTW"
+    # Monkey patch in pyFFTW Numpy interface
```

Comment:
Are you sure that monkey patching into numpy is a good idea? I have no problem to use a better overall structure for the whole module....